src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -343,9 +343,13 @@ protected void extractFile( final File srcF, final File dir, final InputStream c
 
         try
         {
-            if ( !isOverwrite() && f.exists() && ( f.lastModified() >= entryDate.getTime() ) )
+            if ( f.exists() && ( f.lastModified() >= entryDate.getTime() ) )
             {
-                return;
+                String message = String.format( "Archive entry %s already exists on disk and is newer", entryName );
```

Comment:
Why would this indicate a problem? To me it looks like a normal use case when the files on the disk are newer than the ones in the archive (for example when they are generated after the archive is created).

Reply:
It is based on the assumption that you unpack a file into a "clean" directory. If the file on disk is newer than what we are currently extracting, it was written during the same archive extraction. This would mean the archive has entries that (on the users file system, which may be case insensitive) are duplicate.

If you feel that " is newer" is a legitimate situation, I would suggest to at least log the first part ("Archive entry already exists on disk") to inform that some files will be overwritten.

Reply:
Usually the builds are not executed on clean directory. I doubt the usefulness of the message if it appears during normal builds on case sensitive file systems - probably it will be ignored as something normal even when the use case with the case insensitive file systems is hit.

Also on second thought - why does it matter which is newer - the file or the archive entry?

I think it would be best if we can check if the file system is case sensitive(probably not very reliable so maybe we should do it for all  file systems) and then try to determine if the existing file and the archive entry are the same file with different case and then issue a warning.

Reply:
What kind of check would you propose? I would guess if the file name in the archive already exists on disk (case insensitive if filesystem is case insensitive, case sensitive otherwise), we should at least check the size in bytes. But that alone doesn't mean the file is a duplicate.

Or would you suggest to also look at a hash of the file? I think that could have some impact on performance, so I guess we should only do that if the other checks return true.

Reply:
perhaps a simple check on whether the `canonicalDestPath` and the entry's absolute path are the same ? it is a very naive approach , but found it to work well for macOs , there are obviously all the edge cases of symlinks to worry about.

I agree whether the file is newer or not shouldn't matter

And yes, warnings are easily dismissed, but "did you check all the warnings too" is what I tell my team before I burden myself with a context switch :)

Reply:
I expect a release of plexus archiver soonish, would be great to reach consensus here.

Reply:
So, following the proposal of @viqueen, how about a check if the file already exists, regardless of which one is newer. Would you be OK with that, @plamentotev?

Reply:
Hi, sorry for the delay. Would a check if the file name and the entry name differ only by the letter case (string not equals but ignore case equals) work in you case? Then we can log an warning as this should be rare enough and indeed may indicate possible problem on case insensitive file systems. Sorry if that was already discussed.

Also maybe we can wake the warning message more detailed to indicate what the possible issue is. Something like:
"Archive entry %s and existing file %s names differ only by letter case. This may cause issues on case insensitive file systems."

>And yes, warnings are easily dismissed, but "did you check all the warnings too" is what I tell my team before I burden myself with a context switch :)

Is this based on the assumption that the archive will be extracted in clean directory (either fresh CI/CS build or `maven clean verify`)?. This is not always the case  (especially for larger projects where `clean` means a lot of work done again - such as compilation and code generation. But even for small projects constantly cleaning is annoying as it takes some time). As a part of the build a large archive is extracted - it could easily contains thousands of files. Now imagine checking  thousands of warnings on every single build (as there is no actionable item here to fix the warning). I doubt anyone will do it. And even if it is not large file it still could easily contains hundreds of files - still too much. JDK 15 is 441 files on windows - imagine going over each of those 441 files and comparing if the case matches or not. Given the very rare use case of archive containing files that differ only the their letter case for me the annoyance that we introduce will greatly diminish any possible gains. If we give the warning for every single file that is.

On other hand if we only give the warning in specific cases (even with false positives, but given they are rare enough) that could be really helpful. Having a warning if the files differ only by case could be really helpful. Actually maybe even more so when archiving as it would indicate that the resulting archive may not work on all systems.

Reply:
Sure, that warning message would work for me. I think the most important part is that we somehow let the user know that there *may* be a problem.

I will change the code so it logs the message when the entry has a name that only differs by case from the file on disk, regardless of the timestamp of the archive or the file on disk.

Reply:
@plamentotev, please let me know what you think of the updated check & message.

Reply:
@mthmulders apart from the comments the others already pointed out, LGTM. Thanks.